MoTrPAC/motrpac-frontend
# Connects to #95. Node.js, CRA, Auth0, and Storybook version upgrades.

### Technical Notes

**Node.js** `v14.15.4` is required to be running in your local environment when reviewing/testing this PR.

### Key Changes

1. Updated **Circle CI** config to use `node:14.15.4-browsers` docker image.
2. Upgraded **react-scripts** and **react** to latest versions to address vulnerabilities.
3. Bumped versions of **Auth0** and **Storybook** dependencies, along with others.
4. Updated code to comply with **Eslint** rules, while added other **Eslint** rules to temporarily suppress "failed to compile" issues upon upgrading CRA.
5. Updated unit test assertions due to failing tests as a result of version upgrades of node, react, and jest.

### Steps to test

1. Upgrade **Node.js** to `v14.15.4` or use `nvm` to install this version.
2. Clone this PR branch and run the app using `yarn inspect-branch`.
3. Run `yarn audit` in the app's root directory and expect to see the `0 vulnerabilities found` message.
4. Run `yarn test` in the app's root directory and expect to see all tests to pass.
5. Sign in through Auth0 and expect to be able to login successfully.
